models/Import/Service/ResultImporter.php

```diff
@@ -61,34 +61,52 @@ public function importByResultInput(ImportResultInput $input): void
             function () use ($resultStorage, $input, $testUri, $deliveryExecutionUri, $testScoreVariables): void {
                 $this->updateItemResponseVariables($resultStorage, $input, $testUri);
 
-                $updatedScoreTotal = $this->updateItemOutcomeVariables(
+                $totalScoreCalculatedByItemOutcomes = $this->updateItemOutcomeVariables(
                     $resultStorage,
                     $input,
                     $testUri,
-                    $testScoreVariables['scoreTotalMax'],
-                    $testScoreVariables['scoreTotal']
+                    $testScoreVariables['scoreTotal'] ?? 0
                 );
 
+                if (null === $testScoreVariables) {
+                    return;
+                }
+
                 $this->updateTestVariables(
                     $resultStorage,
                     $testScoreVariables['scoreTotalVariable'],
                     $testScoreVariables['scoreTotalVariableId'],
                     $deliveryExecutionUri,
                     $testUri,
-                    $updatedScoreTotal
+                    $totalScoreCalculatedByItemOutcomes,
+                    $testScoreVariables['scoreTotalMax']
                 );
             }
         );
     }
 
+    /**
+     * @throws ImportResultException
+     */
     private function updateTestVariables(
         AbstractRdsResultStorage $resultStorage,
         taoResultServer_models_classes_Variable $scoreTotalVariable,
         int $scoreTotalVariableId,
         string $deliveryExecutionUri,
         string $testUri,
-        float $updatedScoreTotal
+        float $updatedScoreTotal,
+        float $scoreTotalMax
     ): void {
+        if (!empty($scoreTotalMax) && $updatedScoreTotal > $scoreTotalMax) {
+            throw new ImportResultException(
+                sprintf(
+                    'SCORE_TOTAL cannot be higher than SCORE_TOTAL_MAX:%s, %s provided',
```

Comment:
At least for me is better to read :) Wdyt?
```suggestion
                    'SCORE_TOTAL cannot be higher than SCORE_TOTAL_MAX: %s, %s provided',
```